drivers/dma/dma_gd32.c

```diff
@@ -0,0 +1,625 @@
+/*
+ * Copyright (c) 2022 TOKITA Hiroshi <tokita.hiroshi@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT gd_gd32_dma
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/dma.h>
+#include <zephyr/logging/log.h>
+
+#include <gd32_dma.h>
+
+#if defined(GD32F450) || defined(GD32F405) || defined(GD32F407) || defined(GD32F470) ||            \
+	defined(GD32F425) || defined(GD32F427)
+#define GD32F4XX
+#endif
+
+#if defined(GD32F4XX)
+#define CHXCTL_PERIEN_OFFSET ((uint32_t)25U)
+#define GD32_DMA_CHXCTL_DIR  BIT(6)
+#define GD32_DMA_CHXCTL_M2M  BIT(7)
+#define GD32_DMA_FLAG_ERRORS (DMA_FLAG_SDE | DMA_FLAG_TAE)
+#else
+#define GD32_DMA_CHXCTL_DIR  BIT(4)
+#define GD32_DMA_CHXCTL_M2M  BIT(14)
+#define GD32_DMA_FLAG_ERRORS DMA_FLAG_ERR
+#endif
+
+#if defined(GD32F3x0)
+#define GD32_DMA_INTF(dma)	  DMA_INTF
```

Comment:
This line is macro define.
I think it may not cause warning.
Could you show me warning messge?

Reply:
This macro will effect the user, since it receive `dma` as input, they tend to belive that `dma` have been used. But after the preprocessor, there nothing left for the compiler.

Reply:
I meet the same problem at gd32 adc driver. https://github.com/zephyrproject-rtos/zephyr/blob/6ede1da43494da3840b2f270de955f6bf11cbbf0/drivers/adc/adc_gd32.c#L43

Reply:
In SOC_SERIES_GD32F3X0 case,
undefine DMA_INTF, DMA_INTC, DMA_CHCTL, DMA_CHCNT, DMA_CHPADDR, and DMA_CHMADDR, and define same as other SoC models.